controllers/operators/operators_objects.go

```diff
@@ -0,0 +1,206 @@
+func lokiSubscription(name string, namespace string) *ov1alpha1.Subscription {
+	s := subscription(name, constants.LokiOperatorNamespace)
+	s.Spec = &ov1alpha1.SubscriptionSpec{
+		CatalogSource:          constants.ObservabilityName,
+		CatalogSourceNamespace: namespace,
+		Package:                "loki-operator",
+		Channel:                "tech-preview",
+		StartingCSV:            "loki-operator.v0.0.1",
+		InstallPlanApproval:    "Automatic",
+	}
+	return s
+}
+
+func kafkaInstance(ns string) *sv1b2.Kafka {
+	// Default kafka instance configuration
+	version := "3.0.0"
+	config := map[string]interface{}{
+		"offsets.topic.replication.factor":         3,
+		"transaction.state.log.replication.factor": 3,
+		"transaction.state.log.min.isr":            2,
+		"log.message.format.version":               "3.0",
+		"inter.broker.protocol.version":            "3.0",
+	}
+	configStr, err := json.Marshal(config)
+
+	if err != nil {
+		fmt.Printf("can't marshal kafka config %s", err.Error())
+	}
+
+	return &sv1b2.Kafka{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       constants.KafkaName,
+			APIVersion: "kafka.strimzi.io/v1beta2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.KafkaName,
+			Namespace: ns,
+		},
+		Spec: &sv1b2.KafkaSpec{
+			Kafka: sv1b2.KafkaSpecKafka{
+				Config: &apiextensions.JSON{
+					Raw: configStr,
+				},
+				Listeners: []sv1b2.KafkaSpecKafkaListenersElem{
+					{
+						Name: "plain",
+						Port: 9092,
+						Tls:  false,
+						Type: "internal",
+					},
+					{
+						Name: "tls",
+						Port: 9093,
+						Tls:  true,
+						Type: "internal",
+					},
+				},
+				Replicas: 3,
+				Storage: sv1b2.KafkaSpecKafkaStorage{
+					Type: "ephemeral",
+				},
+				Version: &version,
+			},
+			Zookeeper: sv1b2.KafkaSpecZookeeper{
+				Replicas: 3,
+				Storage: sv1b2.KafkaSpecZookeeperStorage{
+					Type: "ephemeral",
+				},
+			},
+		},
+	}
+}
+
+func grafanaInstance(ns string) *gv1alpha1.Grafana {
+	return &gv1alpha1.Grafana{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       constants.GrafanaName,
+			APIVersion: "integreatly.org/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.GrafanaName,
+			Namespace: ns,
+		},
+		Spec: gv1alpha1.GrafanaSpec{
+			Config: gv1alpha1.GrafanaConfig{
+				Security: &gv1alpha1.GrafanaConfigSecurity{
+					AdminUser:     "admin",
+					AdminPassword: "admin",
+				},
+			},
+		},
+	}
+}
+
+func lokiInstance() *lv1beta1.LokiStack {
+	return &lv1beta1.LokiStack{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       constants.LokiName,
+			APIVersion: "integreatly.org/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.LokiName,
+			Namespace: constants.LokiOperatorNamespace,
+		},
+		Spec: lv1beta1.LokiStackSpec{
+			Size:              lv1beta1.SizeOneXExtraSmall,
+			ReplicationFactor: 1,
+			Storage: lv1beta1.ObjectStorageSpec{
+				Secret: lv1beta1.ObjectStorageSecretSpec{
+					Name: "test",
+				},
+			},
+			StorageClassName: "standard",
```

Comment:
I'm not sure about that "standard" storage class name since I had it working on `cluster bot` and not on `openshift-install`

Anyway if you have `Pending` loki components you can either change that value in the code or create a "standard" storage class to the storage type you want like:
```yaml
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: standard
provisioner: kubernetes.io/aws-ebs
parameters:
  type: gp2
reclaimPolicy: Retain
allowVolumeExpansion: true
volumeBindingMode: Immediate

```

Reply:
@jpinsonneau maybe you can add this "as code" - this is similar to https://raw.githubusercontent.com/netobserv/documents/main/examples/zero-click-loki/1-storage.yaml to some level.